### ISSUE TITLE
Crash in `FlutterViewController.updateSemantics`

### ISSUE THREAD

**Issue by tvolkert** [flutter/flutter#152828]
## Steps to reproduce:

1. Run the following app:
   ```dart
   import 'package:flutter/material.dart';

   void main() {
     runApp(const MaterialApp(
       home: BugReport(),
       title: 'Bug Report',
     ));
   }

   class BugReport extends StatelessWidget {
     const BugReport({super.key});

     static const List<Color> colors = <Color>[
       Colors.red,
       Colors.blue,
     ];

     @override
     Widget build(BuildContext context) {
       return Scaffold(
         body: Column(
           crossAxisAlignment: CrossAxisAlignment.stretch,
           children: <Widget>[
             Expanded(
               child: ColoredBox(
                 color: Colors.yellow,
                 child: GestureDetector(
                   onTap: () => BugReportDialog.show(context),
                 ),
               ),
             ),
             SizedBox(
               height: 175,
               child: ListView.builder(
                 itemExtent: 175,
                 scrollDirection: Axis.horizontal,
                 itemCount: 2,
                 itemBuilder: (BuildContext context, int index) => ColoredBox(color: colors[index]),
               ),
             ),
           ],
         ),
       );
     }
   }

   class BugReportDialog extends StatefulWidget {
     const BugReportDialog({super.key});

     static Future<void> show(BuildContext context) {
       return showDialog<void>(
         context: context,
         builder: (BuildContext context) => const BugReportDialog(),
       );
     }

     @override
     State<BugReportDialog> createState() => _BugReportDialogState();
   }

   class _BugReportDialogState extends State<BugReportDialog> {
     bool selected = false;

     void _handleSelect(bool? value) {
       setState(() {
         selected = value!;
       });
     }

     @override
     Widget build(BuildContext context) {
       return Material(
         child: Row(
           children: <Widget>[
             Radio<bool>(
               value: true,
               groupValue: selected,
               onChanged: _handleSelect,
             ),
             const Text('Select the radio, then type the minus sign here: '),
             Expanded(
               child: TextField(
                 enabled: selected,
                 autofocus: false,
                 maxLength: 12,
               ),
             ),
           ],
         ),
       );
     }
   }
   ```
1. Click on the blue box
1. CMD-click on the red box
1. Click on the yellow box
1. Select the radio button
1. Click on the text input to focus it
1. Type the minus ("-") sign using your physical keyboard 

## Expected behavior

You expect to be able to type in the text input with no issues

## Actual behavior

The app crashes with the following trace:

```
*** Terminating app due to uncaught exception 'NSInvalidArgumentException', reason: '-[AXPlatformNodeCocoa startEditing]: unrecognized selector sent to instance 0x600003bddac0'
*** First throw call stack:
(
	0   CoreFoundation                      0x0000000193d832ec __exceptionPreprocess + 176
	1   libobjc.A.dylib                     0x000000019386a788 objc_exception_throw + 60
	2   CoreFoundation                      0x0000000193e3556c -[NSObject(NSObject) __retain_OA] + 0
	3   CoreFoundation                      0x0000000193cecf3c ___forwarding___ + 1580
	4   CoreFoundation                      0x0000000193cec850 _CF_forwarding_prep_0 + 96
	5   FlutterMacOS                        0x00000001061e1c04 _ZNK7flutter22AccessibilityBridgeMac22MacOSEventsFromAXEventEN2ui16AXEventGenerator5EventERKNS1_6AXNodeE + 1216
	6   FlutterMacOS                        0x00000001061e1604 _ZN7flutter22AccessibilityBridgeMac20OnAccessibilityEventEN2ui16AXEventGenerator13TargetedEventE + 196
	7   FlutterMacOS                        0x0000000106a61014 _ZN7flutter19AccessibilityBridge13CommitUpdatesEv + 1584
	8   FlutterMacOS                        0x000000010621724c -[FlutterViewController updateSemantics:] + 208
	9   FlutterMacOS                        0x00000001061f95dc _ZZ35-[FlutterEngine runWithEntrypoint:]EN3$_18__invokeEPK23FlutterSemanticsUpdate2Pv + 88
	10  FlutterMacOS                        0x0000000106a738c0 _ZNSt3_fl10__function6__funcIZ39CreateEmbedderSemanticsUpdateCallbackV3PFvPK23FlutterSemanticsUpdate2PvES5_E3$_0NS_9allocatorIS8_EEFvNS_13unordered_mapIiN7flutter13SemanticsNodeENS_4hashIiEENS_8e	11  FlutterMacOS                        0x0000000106a864b0 _ZN7flutter20PlatformViewEmbedder15UpdateSemanticsENSt3_fl13unordered_mapIiNS_13SemanticsNodeENS1_4hashIiEENS1_8equal_toIiEENS1_9allocatorINS1_4pairIKiS3_EEEEEENS2_IiNS_25CustomAccessibilityActio	12  FlutterMacOS                        0x0000000106992428 _ZNSt3_fl10__function6__funcIZN7flutter5Shell23OnEngineUpdateSemanticsENS_13unordered_mapIiNS2_13SemanticsNodeENS_4hashIiEENS_8equal_toIiEENS_9allocatorINS_4pairIKiS5_EEEEEENS4_IiNS2_25CustomAcce	13  FlutterMacOS                        0x0000000106a83bb0 _ZN7flutter18EmbedderTaskRunner8PostTaskEy + 652
	14  FlutterMacOS                        0x0000000106a6c598 FlutterEngineRunTask + 36
	15  FlutterMacOS                        0x00000001061f8e40 -[FlutterEngine runTaskOnEmbedder:] + 56
	16  FlutterMacOS                        0x00000001061f8ff4 __60-[FlutterEngine postMainThreadTask:targetTimeInNanoseconds:]_block_invoke + 64
	17  libdispatch.dylib                   0x0000000193a7c750 _dispatch_call_block_and_release + 32
	18  libdispatch.dylib                   0x0000000193a7e3e8 _dispatch_client_callout + 20
	19  libdispatch.dylib                   0x0000000193a8cbb8 _dispatch_main_queue_drain + 988
	20  libdispatch.dylib                   0x0000000193a8c7cc _dispatch_main_queue_callback_4CF + 44
	21  CoreFoundation                      0x0000000193d4fad4 __CFRUNLOOP_IS_SERVICING_THE_MAIN_DISPATCH_QUEUE__ + 16
	22  CoreFoundation                      0x0000000193d0d258 __CFRunLoopRun + 1996
	23  CoreFoundation                      0x0000000193d0c434 CFRunLoopRunSpecific + 608
	24  HIToolbox                           0x000000019e4b019c RunCurrentEventLoopInMode + 292
	25  HIToolbox                           0x000000019e4affd8 ReceiveNextEventCommon + 648
	26  HIToolbox                           0x000000019e4afd30 _BlockUntilNextEventMatchingListInModeWithFilter + 76
	27  AppKit                              0x000000019756bd68 _DPSNextEvent + 660
	28  AppKit                              0x0000000197d61808 -[NSApplication(NSEventRouting) _nextEventMatchingEventMask:untilDate:inMode:dequeue:] + 700
	29  AppKit                              0x000000019755f09c -[NSApplication run] + 476
	30  AppKit                              0x00000001975362e0 NSApplicationMain + 880
	31  geotag                              0x00000001029275f8 $sSo21NSApplicationDelegateP6AppKitE4mainyyFZ + 40
	32  geotag                              0x00000001029275c0 $s6geotag11AppDelegateC5$mainyyFZ + 44
	33  geotag                              0x0000000102927660 main + 28
	34  dyld                                0x00000001938a60e0 start + 2360
)
libc++abi: terminating due to uncaught exception of type NSException
```

## Flutter version

```
[✓] Flutter (Channel main, 3.23.0-14.0.pre.76, on macOS 14.5 23F79 darwin-arm64, locale en)
    • Flutter version 3.23.0-14.0.pre.76 on channel main at /Users/tvolkert/project/flutter/flutter
    • Upstream repository git@github.com:flutter/flutter.git
    • Framework revision 1c9c375e4e (5 weeks ago), 2024-06-30 07:43:26 -0400
    • Engine revision 0d93da7e2f
    • Dart version 3.5.0 (build 3.5.0-318.0.dev)
    • DevTools version 2.37.0
```

**Comment by knopp** [comment]
I think this should be also fixed by https://github.com/flutter/engine/commit/206e86ee8a40ad28233338b7a99a2b083cdce85f.

**Comment by tvolkert** [comment]
Yep - verified it's fixed! 🥳
